data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java

```diff
@@ -569,13 +568,43 @@ public Date getDate(int columnIndex, Calendar calendar) throws SQLException {
                         return null;
                     }
                     String dateString = getString(columnIndex);
-                    FastDateFormat dateFormat;
-                    if (dateString.length() < 21) {
-                        dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss", calendar.getTimeZone());
-                    } else {
-                        dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSS", calendar.getTimeZone());
+                    
+                    // First try the original FastDateFormat approach with strict patterns
+                    try {
```

Comment:
Do you really need all of this?
Does the java datetimeformatter not support Z by itself?

Also there's seem to be support for lenient parsing - https://stackoverflow.com/questions/30090710/java-8-datetimeformatter-parsing-for-optional-fractional-seconds-of-varying-sign

Try messing with that before manually manipulating the string.

Either way, if there's complex logic like this I think it should go to a different method.

Reply:
changed the implementation a bit to use flexible format using a formatter builder. Can you have a look on this @AsafMah